src/uu/rm/src/rm.rs

```diff
@@ -306,7 +307,15 @@ fn remove_dir(path: &Path, options: &Options) -> bool {
                 }
             }
             Err(e) => {
-                show_error!("removing '{}': {}", path.display(), e);
+                if e.to_string().starts_with("Directory not empty") || (cfg!(windows) && e.to_string().starts_with("The directory is not empty")) {
```

Comment:
I don't think we can rely on the error message itself; it might change in rustc itself
there is no stable error code  here?

Reply:
I've tried looking, and I honestly can't find one. Do you have any ideas? I thought that the raw error code would work, but it's different across Windows, Linux, etc. The two error messages given should work for all Unix based operating systems, and Windows is its own beast.

Reply:
It also does pass all of the tests, and any non Unix-based OS would be difficult to plan for, since there error codes could be anything at all.

Reply:
Not sure what to do here tbh :/
@Arcterus @rivy any idea/suggestion ? :)

Reply:
On Unix, it’s probably best to check `errno` for `ENOTEMPTY`/`EEXIST` (you can use either `nix` or `libc` directly).  I’m not exactly sure how to check the windows error code, but there’s likely a function in `winapi` (or the new crate made by Microsoft itself?).